contracts/AnchoredPriceView.sol

```diff
@@ -0,0 +1,331 @@
+            uint256 ethPerToken = anchor.getUnderlyingPrice(getCTokenAddress(symbol));
+
+            // ethPerToken has 18 decimals since the usdt, usdc, wbtc tokens hit
+            // usdPerEth has 6 decimals
+            // scaling by 1e18 and dividing leaves 1e6, as desired
+            return mul(usdPerEth, 1e18) / ethPerToken;
+        }
+    }
+
+    /**
+     * @dev fetch price in eth from proxy and convert to usd price using anchor usdc price.
+     * @dev Anchor usdc price has 30 decimals, and anchor general price has 18 decimals, so multiplying 1e18 by 1e18 and dividing by 1e30 yields 1e6
+     */
+    function getAnchorPrice(address tokenAddress, uint256 usdcPrice) public view returns (uint256) {
+        if ( tokenAddress == cUsdcAddress || tokenAddress == cUsdtAddress )  {
+            // hard code to 1 dollar
+            return oneDollar;
+        }
+
+        uint priceInEth = anchor.getUnderlyingPrice(tokenAddress);
+        uint additionalScale;
+        if ( tokenAddress == cWbtcAddress ){
+            // wbtc proxy price is scaled 1e(36 - 8) = 1e28, so we need 8 more to get to 36
+            additionalScale = 1e8;
+        } else {
+            // all other tokens are scaled 1e18, so we need 18 more to get to 36
+            additionalScale = 1e18;
+        }
+
+        // usdcPrice has 30 decimals, so final result has 6
+        return mul(priceInEth, additionalScale) / usdcPrice;
+    }
+
+    /**
+     * @notice Implements the method of the PriceOracle interface of Compound v2 and returns returns the Eth price for an asset.
+     * @dev converts from 1e6 decimals of Open Oracle to 1e(36 - underlyingDecimals) of PriceOracleProxy
+     * @param cToken The cToken address for price retrieval
+     * @return The price for the given cToken address
+     */
+    function getUnderlyingPrice(address cToken) external view returns (uint256) {
+        if (breaker == true) {
+            return anchor.getUnderlyingPrice(cToken);
+        }
+
+        uint256 usdPerToken = _prices[getOracleKey(cToken)];
+
+        if ( usdPerToken == 0 ) {
+            return anchor.getUnderlyingPrice(cToken);
+        } else {
+            uint256 usdPerEth = _prices["ETH"];
+            uint256 ethPerToken = mul(usdPerToken, 1e6) / usdPerEth;
+            uint256 additionalScale = getAdditionalScale(cToken);
+
+            return mul(ethPerToken, additionalScale);
+        }
+    }
+
+    /**
+     * comptroller expects price to have 18 decimals,
+     * additionally upscaled by 1e18 - underlyingdecimals
+     * base decimals is 1e6, so start by addint twelve
+     */
+    function getAdditionalScale(address cToken) public view returns (uint256) {
+        // total scale 1e30
+        if (cToken == cUsdcAddress) return 1e24;
+        if (cToken == cUsdtAddress) return 1e24;
+        // total scale 1e28
+        if (cToken == cWbtcAddress) return 1e22;
+        // total scale 1e18
+        if (cToken == cEthAddress) return 1e12;
```

Comment:
what about scale for other tokens?

Reply:
they aren't being posted by coinbase right now, but I think you are right maybe we can just return 1e12 for the rest of them ( they all have 18 decimals ) and this will automatically work if coinbase adds them. I was kind of in a "we'll have to change the view when they add assets" mindset, but that may not be true